errors/context.go

```diff
@@ -0,0 +1,42 @@
+package errors
+
+type errorWithContext interface {
```

Comment:
This name is slightly misleading, since `errorWithContext` isn't actually an `error`.